core/src/main/scala/akka/kafka/scaladsl/Consumer.scala

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <http://softwaremill.com>
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.kafka.scaladsl
+
+import scala.concurrent.Future
+import akka.Done
+import akka.stream.scaladsl.Source
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import org.apache.kafka.clients.consumer.KafkaConsumer
+import akka.kafka.internal.ConsumerStage
+import akka.NotUsed
+import akka.dispatch.ExecutionContexts
+import scala.concurrent.ExecutionContext
+import org.apache.kafka.common.TopicPartition
+import org.apache.kafka.clients.consumer.OffsetAndMetadata
+import akka.kafka.internal.ConsumerStage.CommittableOffsetBatchImpl
+import akka.kafka.internal.PlainConsumerStage
+import akka.kafka.internal.CommittableConsumerStage
+
+object Consumer {
+
+  trait Message[K, V] {
+    def key: K
+    def value: V
+    def offset: PartitionOffset
+  }
+
+  final case class CommittableMessage[K, V](
+      key: K,
+      value: V,
+      committableOffset: CommittableOffset
+  ) extends Message[K, V] {
+
+    override def offset: PartitionOffset = committableOffset.offset
+  }
+
+  /**
+   * This interface might move into `akka.stream`
+   */
+  trait Committable {
+    def commit(): Future[Done]
+  }
+
+  trait CommittableOffset extends Committable {
+    def offset: PartitionOffset
+  }
+
+  final case class PartitionOffset(topic: String, partition: Int, offset: Long)
+
+  object CommittableOffsetBatch {
```

Comment:
> Why this code will not work (concept):

Is that a question?

commit for topic1 and topic2 could be grouped together if they originated from the same stage. That would be more efficient than committing topic1 and topic2 individually, as in your example. Is my assumption of that wrong or is it not needed?